src/cmd/usb-util.js

```diff
@@ -153,12 +203,15 @@ async function getOneUsbDevice({ idOrName, api, auth, ui }) {
 			name: 'device',
 			message: 'Which device would you like to select?',
 			choices() {
-				return usbDevices.map((d) => {
+				return Promise.all(usbDevices.map(async (d) => {
+					const id = await _getDeviceId(d);
+					const mode = await _getDeviceMode(d);
```

Comment:
It might be good if we open the device once and get the mode and the id in one step?
Also that will allow us to have a `parallel` behavior here:

```
[{id, mode}, name] = Promise.all[_getDeviceInfo(d), _getDeviceName({...})]

```

Reply:
Good suggestion

Reply:
@hugomontero Added a `_getDeviceInfo` instead of two different methods. But to write everything in a single Promise.all() , the `_getDeviceName()` requires `id` which is available from the first call. Let me know if I am missing something.

Reply:
Is like you said hehe. 👍